File: android/src/main/kotlin/br/com/bluethermal/blue_thermal_plus/core/PrinterTransportManager.kt
package br.com.bluethermal.blue_thermal_plus.core

interface PrinterTransportManager {
    var onEvent: ((Map<String, Any>) -> Unit)?

    fun startScan()
    fun stopScan()
    fun connect(deviceId: String)
    fun disconnect()
    fun printRaw(data: ByteArray)
}

File: ios/Classes/core/PrinterTransportManager.swift
import Foundation

protocol PrinterTransportManager: AnyObject {
  /// Emits events to Flutter as a Dictionary payload.
  /// Example:
  /// { "type": "deviceFound", "device": { "id": "...", "name": "..." }, "message": "..." }
  var onEvent: (([String: Any]) -> Void)? { get set }

  func startScan()
  func stopScan()

  func connect(deviceId: String)
  func disconnect()

  func printRaw(data: Data)
}
